pkg/reconciler/scheduling/edgeplacement/controller.go

```diff
@@ -183,68 +133,21 @@ func (c *controller) enqueuePlacement(obj interface{}) {
 	c.queue.Add(key)
 }
 
-// enqueueNamespace enqueues all placements for the namespace.
-func (c *controller) enqueueNamespace(obj interface{}) {
-	logger := logging.WithReconciler(klog.Background(), ControllerName)
-	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-	clusterName, _, _, err := kcpcache.SplitMetaClusterNamespaceKey(key)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	placements, err := c.placementLister.Cluster(clusterName).List(labels.Everything())
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	for _, placement := range placements {
-		namespaceKey := key
-		key, err := kcpcache.MetaClusterNamespaceKeyFunc(placement)
-		if err != nil {
-			runtime.HandleError(err)
-			continue
-		}
-		logging.WithQueueKey(logger, key).V(2).Info("queueing Placement because Namespace changed", "namespace", namespaceKey)
-		c.queue.Add(key)
-	}
-}
-
 func (c *controller) enqueueLocation(obj interface{}) {
-	logger := logging.WithReconciler(klog.Background(), ControllerName)
-	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-	clusterName, _, _, err := kcpcache.SplitMetaClusterNamespaceKey(key)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	placements, err := c.placementIndexer.ByIndex(byLocationWorkspace, clusterName.String())
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	for _, obj := range placements {
-		placement := obj.(*schedulingv1alpha1.Placement)
-		locationKey := key
-		key, err := kcpcache.MetaClusterNamespaceKeyFunc(placement)
-		if err != nil {
-			runtime.HandleError(err)
-			continue
-		}
-		logging.WithQueueKey(logger, key).V(2).Info("queueing Placement because Location changed", "location", locationKey)
-		c.queue.Add(key)
-	}
+	// logger := logging.WithReconciler(klog.Background(), ControllerName)
+	// key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(obj)
+	// if err != nil {
+	// 	runtime.HandleError(err)
+	// 	return
+	// }
+	// clusterName, _, _, err := kcpcache.SplitMetaClusterNamespaceKey(key)
+	// if err != nil {
+	// 	runtime.HandleError(err)
+	// 	return
+	// }
+
+	// TODO: We may or may not need to enqueue edgeplacements because of changes from locations.
+	// That depends on our future design.
```

Comment:
This process should react to Location changes as well as Placement changes.  Perhaps use two workqueues, one for each kind of object, or use one workqueue with object kind included in the object reference values put into the queue.  Or, as does the old code here, use just one queue that holds only references to placements and bluntly enqueue references to all that could matter.

Reply:
Changed the comment to documents this. Thanks for suggesting these approaches!